packages/server/test/interop/react/invalidateQueries.test.tsx

```diff
@@ -43,8 +43,8 @@ describe('invalidateQueries()', () => {
           <button
             data-testid="refetch"
             onClick={() => {
-              queryClient.invalidateQueries(['allPosts']);
-              queryClient.invalidateQueries(['postById']);
+              queryClient.invalidateQueries([['allPosts']]);
+              queryClient.invalidateQueries([['postById']]);
```

Comment:
This doesn't look quite right? Shouldn't it be unchanged?

Reply:
> The only breaking change from this is if someone is delving in and using the query keys not through the tRPC provided API's but by using Tanstack Query's queryClient directly. In this case users will see keys change from something like
> ["post. List"] to [["post","list"]]
> 
> An example of the breaking change can be seen in the test I had to change in this PR.

👆This was what this comment in the PR description is about 🙂

----------

We are now always storing the `path` as a sub array as the first element of the `queryKey`. 

The initial conversation around whether it should always be a sub array or top level elements is [here in the issue](https://github.com/trpc/trpc/issues/2611#issuecomment-1237490730), but we decided to go with the sub array direction so that the `path` is always kept neatly as the "first element" of the key! @TkDodo [confirmed there is no functional difference](https://github.com/trpc/trpc/issues/2611#issuecomment-1244181242) in terms of invalidation. 

This doesn't effect usage any where if you are using the tRPC provided API's but in this test we are delving in and getting the tanstack `queryClient` directly... And so the keys would be changed from what they were before! 

For a deeper defined procedure this is a bit more evident and would break this test even if it was a sub array or not!: `queryClient.invalidateQueries(['posts.all']);` becomes `queryClient.invalidateQueries([['posts','all']]);`

----------
Does that make sense? what do you think?

Reply:
if users have access to the queryClient directly as well, it might be better to not do the nested arrays.

Reply:
I'm still personally a fan of the nested array (can you guess? 😂) just so that array keys are always a nice touple of `[path[], input]` which if we do them at the top level they won't be it would be more `[path1, path2, path3, input]`

If we did make them top level this test would technically not have to change to pass... however, its a simplified edge case example as it only has one level of the path... `queryClient.invalidateQueries(['posts. All']);` would still become `queryClient.invalidateQueries(['posts.','all']);` and break when working directly with the `queryClient`! 

🤔💭